s3/pkg/datastore/yig/storage/object.go

```diff
@@ -128,6 +128,22 @@ func (yig *YigStorage) Put(ctx context.Context, stream io.Reader, obj *pb.Object
 		userMd5 = val.(string)
 	}
 
+	// check and remove the old object if exists.
+	if obj.StorageMeta != "" && obj.ObjectId != "" {
+		storageMeta, err := ParseObjectMeta(obj.StorageMeta)
+		if err != nil {
+			log.Errorf("Put(%s, %s, %s) failed, failed to parse storage meta(%s), err: %v", obj.BucketName,
+				obj.ObjectKey, obj.ObjectId, obj.StorageMeta, err)
+			return result, err
+		}
+		err = yig.putObjToGc(storageMeta.Cluster, storageMeta.Pool, obj.ObjectId)
```

Comment:
Hi @hopkings2008 , if versioning enabled, does different version of the same object have different objectId or not?

Reply:
Yes, if version is enabled, there are different oids for different versions. but the storage backend cannot see the version, it only handles oids. version is handled in upper layer.

Reply:
OK, thanks.

Reply:
just transfer both the storage meta(transfer it even if it may be "", the backend will handle it) and object id to the backend.